### PR TITLE
Stereo

### LINKED ISSUE #53: calibrate and generate disparity and distance maps from stereo cameras

### PR DESCRIPTION
**Pull request by pgayee694**
This is the stereo calibration and distance calculations

### CODE REVIEW

**Comment by pgayee694** [comment]
Thinking about how to write tests for some of this stuff, specifically on what I'll need to mock and how, but would appreciate review on the other stuff in the mean time

**Review by mooncollin — APPROVED** [review]
We might need to discuss how often/when calibration should happen.

**Comment by mooncollin** [comment]
Also what's going on with the conflicts? From the web view, it seems like the requirements.txt file is being blanked.

File: seers/calibration/calibrate.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+        ret, framel = videol.retrieve()
+        ret, framer = videor.retrieve()
+        grayl = cv2.cvtColor(framel, cv2.COLOR_BGR2GRAY)
+        grayr = cv2.cvtColor(framer, cv2.COLOR_BGR2GRAY)
+
+        #cv2.imshow('framel', framel)
+        #cv2.imshow('framer', framer)
+
+        #cv2.waitKey(1)
```
Comment (svnphlps) [inline seers/calibration/calibrate.py]: remove

File: seers/calibration/calibrate.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+            while now - start < 5:
+                # avoid the video advancing when we don't want it to
+
+                framel = videol.read()
+                framer = videor.read()
+                #cv2.imshow('left', framel)
+                #cv2.imshow('right', framer)
+
+                #cv2.waitKey(1)
```
Comment (svnphlps) [inline seers/calibration/calibrate.py]: remove

File: seers/calibration/calibrate.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    
+    videol.release()
+    videor.release()
+    cv2.destroyAllWindows()
+else:
+    # We already have an image directory
+
+    for imagel, imager in zip(imagesl, imagesr):
+        imgl = cv2.imread(imagel)
+        #grayl = cv2.cvtColor(imgl, cv2.COLOR_BGR2GRAY)
```
Comment (svnphlps) [inline seers/calibration/calibrate.py]: remove

File: seers/calibration/calibrate.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+        imgr = cv2.imread(imager)
+        #grayr = cv2.cvtColor(imgr, cv2.COLOR_BGR2GRAY)
```
Comment (svnphlps) [inline seers/calibration/calibrate.py]: remove

File: seers/calibration/calibrate.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+mapXr, mapYr = cv2.initUndistortRectifyMap(mtxr, distr, rectifr, projr, (1280, 720), cv2.CV_32FC1)
+
+framel = cv2.imread('./images/chessboard-5l.jpg')
+framer = cv2.imread('./images/chessboard-5r.jpg')
+
+undistorted_rectifiedl = cv2.remap(framel, mapXl, mapYl, cv2.INTER_LINEAR)
+undistorted_rectifiedr = cv2.remap(framer, mapXr, mapYr, cv2.INTER_LINEAR)
+
+#cv2.imshow('rectifiedl', undistorted_rectifiedl)
+#cv2.imshow('rectifiedr', undistorted_rectifiedr)
```
Comment (svnphlps) [inline seers/calibration/calibrate.py]: remove

File: seers/calibration/calibrate.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+cv2.imwrite('./generated/undistorted_rectifiedl.jpg', undistorted_rectifiedl)
+cv2.imwrite('./generated/undistorted_rectifiedr.jpg', undistorted_rectifiedr)
+
+#cv2.waitKey(0)
```
Comment (svnphlps) [inline seers/calibration/calibrate.py]: remove

File: seers/calibration/calibrate.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+    videol = cv2.VideoCapture(0)
+    videol.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
+    videol.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)
+
+    videor = cv2.VideoCapture(1)
+    videor.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
+    videor.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)
+
+    #videol.set(cv2.CAP_PROP_AUTOFOCUS, 0)
```
Comment (svnphlps) [inline seers/calibration/calibrate.py]: remove

**Review by dbrink1108 — DISMISSED** [review]

**Review by svnphlps — APPROVED** [review]

**Pull request merged**